releases/release-4.0-ga.md

```diff
@@ -0,0 +1,114 @@
+
+# TiDB 4.0 GA Release Notes
+
+Release date: May 28, 2020
+
+TiDB version: 4.0.0
+
+## Compatibility Changes
+
+* TiDB
+    - Optimize the error message of the large-sized transactions for easier troubleshooting [#17219](https://github.com/pingcap/tidb/pull/17219)
+
+* TiCDC
+    - Optimize the structure of the `Changefeed` configuration file to improve usability [#588](https://github.com/pingcap/ticdc/pull/588)
+    - Add the `ignore-txn-start-ts` configuration item, and change the condition from `commit_ts` to `start_ts` during transactions filtering [#589](https://github.com/pingcap/ticdc/pull/589)
+
+## Important Bug Fixes
+
+* TiKV
+    - Fix the `DefaultNotFound` error that occurs when backing up using Backup & Restore (BR) [#7937](https://github.com/tikv/tikv/pull/7937)
+    - Fix system panics caused by the out-of-order `ReadIndex` packages [#7930](https://github.com/tikv/tikv/pull/7930)
+    - Fix system panics caused by incorrectly removing snapshot files after TiKV is restarted [#7927](https://github.com/tikv/tikv/pull/7927)
+
+* TiFlash
+    - Fix the possible data loss issue that occurs when the system panics because of incorrect processing logic of `Raft Admin Command`
+
+## New Features
+
+* TiDB
+    - Add the `committer-concurrency` configuration item to control the number of `goroutines` in the retry-commit phase [#16849](https://github.com/pingcap/tidb/pull/16849)
+    - Support the `show table partition regions` syntax [#17294](https://github.com/pingcap/tidb/pull/17294)
+    - Add the `tmp-storage-quota` configuration item to limit the temporary disk space used by the TiDB server [#15700](https://github.com/pingcap/tidb/pull/15700)
+    - Support checking whether the partitioned table uses a unique prefix index when creating and changing tables [#17213](https://github.com/pingcap/tidb/pull/17213)
+    - Support the `insert/replace into tbl_name partition`(`partition_name_list`) statement [#17313](https://github.com/pingcap/tidb/pull/17313)
+    - Support checking the value of `collations` when using the `Distinct` function [#17240](https://github.com/pingcap/tidb/pull/17240)
+    - Support the `is null` filter condition during the Hash partition pruning [#17310](https://github.com/pingcap/tidb/pull/17310)
+    - Support `admin check index`, `admin cleanup index`, and `admin recover index` in partitioned tables [#17392](https://github.com/pingcap/tidb/pull/17392) [#17405](https://github.com/pingcap/tidb/pull/17405) [#17317](https://github.com/pingcap/tidb/pull/17317)
+    - Support range partition pruning for the `in` expressions  [#17320](https://github.com/pingcap/tidb/pull/17320)
+
+* TiFlash
+    - Support filtering out the data corresponding to the qualified `TSO` through the `min commit ts` value of the `Lock CF` when the Learner reads the data
+    - Add the feature that the system explicitly reports an error to avoid incorrect calculation result when the value of `TIMESTAMP` type is less than `1970-01-01 00:00:00`
```

Comment:
```suggestion
    - Add the feature that the system explicitly reports an error to avoid incorrect calculation results when the value of `TIMESTAMP` type is less than `1970-01-01 00:00:00`
```